oidc/request.go

```diff
@@ -83,6 +83,12 @@ type Request interface {
 	// See: https://tools.ietf.org/html/rfc7636
 	PKCEVerifier() CodeVerifier
 
+	// ClientAssertionJWT optionally specifies a signed JWT to be used in a
+	// client_assertion token request.
+	//
+	// See: https://oauth.net/private-key-jwt/
+	ClientAssertionJWT() string
```

Comment:
Could we make this return something more structured than a string? Since this is an interface, I think it might be nice to make this stronger typed. Could we use the *`clientassertion.ClientAssertion` type?

Reply:
Yeah, I'm going to take a swing at this, because it would make it harder to hold incorrectly.